evap/staff/views.py

```diff
@@ -679,27 +682,37 @@ def semester_import(request, semester_id):
 
 
 @manager_required
-def semester_export(request, semester_id):
-    semester = get_object_or_404(Semester, id=semester_id)
+class SemesterExportView(SingleObjectMixin, FormsetView):
+    model = Semester
+    pk_url_kwarg = "semester_id"
+    form_class = formset_factory(form=ExportSheetForm, can_delete=True, extra=0, min_num=1, validate_min=True)
+    template_name = "staff_semester_export.html"
 
-    ExportSheetFormset = formset_factory(form=ExportSheetForm, can_delete=True, extra=0, min_num=1, validate_min=True)
-    formset = ExportSheetFormset(request.POST or None, form_kwargs={"semester": semester})
+    def dispatch(self, *args, **kwargs):
+        self.object = self.semester = self.get_object()
```

Comment:
We could just always refer to `self.object`, I think I'd be fine with that? Might backfire though. But having two different names might also backfire...

Edit: We use `object` in the `CourseEditView` below, I think having it explicitly declared here as `object: Semester` should be fine.